src/ci_providers/provider_jenkinsci.ts

```diff
@@ -1,20 +1,21 @@
-const { parseSlugFromRemoteAddr } = require('../helpers/git')
+import { parseSlugFromRemoteAddr } from '../helpers/git'
+import { IServiceParams, UploaderEnvs, UploaderInputs } from '../types'
 
-function detect(envs) {
-  return envs.JENKINS_URL
+export function detect(envs: UploaderEnvs): boolean {
+  return Boolean(envs.JENKINS_URL)
 }
 
-function _getBuild(inputs) {
+function _getBuild(inputs: UploaderInputs): string {
   const { args, envs } = inputs
-  return args.build || envs.BUILD_NUMBER
+  return args.build || envs.BUILD_NUMBER || ''
 }
 
-function _getBuildURL(inputs) {
+function _getBuildURL(inputs: UploaderInputs): string {
   const { envs } = inputs
-  return envs.BUILD_URL ? encodeURIComponent(envs.BUILD_URL) : ''
+  return envs.BUILD_URL ? encodeURIComponent(envs.BUILD_URL || '') : ''
```

Comment:
```suggestion
  return envs.BUILD_URL ? encodeURIComponent(envs.BUILD_URL) : ''
```

`BUILD_URL` is truthy in the truthful expression of the ternary operator. In other words, it is guaranteed to not be an empty string when passed to encodeURIComponent.